test/driver/exception_handling/run.t

```diff
@@ -77,47 +79,52 @@ can be reported about it.
   $ echo "let x = 1+1. " > impl.ml
   $ echo "let _ = [%gen_raise_located_error]" >> impl.ml
   $ ./extender.exe -embed-errors impl.ml
-  [%%ocaml.error "A raised located error"]
+  [%%ocaml.error "(context-free ppx pass) A raised located error"]
 
  In the case of derivers
 
   $ echo "let x = 1+1. " > impl.ml
   $ echo "type a = int" > impl.ml
   $ echo "type b = int [@@deriving deriver_located_error]" >> impl.ml
   $ ./deriver.exe -embed-errors impl.ml
-  [%%ocaml.error "A raised located error"]
+  [%%ocaml.error "(context-free ppx pass) A raised located error"]
 
  In the case of whole file transformations:
 
   $ echo "let x = 1+1. " > impl.ml
   $ ./whole_file_located_error.exe -embed-errors impl.ml
-  [%%ocaml.error "A located error in a whole file transform"]
+  [%%ocaml.error
+    "The following located exception was raised during the whole file transformation phase of the ppx \"raise_exc\":\nA located error in a whole file transform\nThe use of located exception is discouraged in whole file transformation as it prevents other errors to be reported. Instead, errors should be embedded in the AST in extension nodes. You might want to file an issue to the ppx authors."]
 
 3. Raising an exception. The exception is not caught by the driver.
 
  In the case of extensions:
 
   $ echo "let _ = [%gen_raise_exc] + [%gen_raise_exc]" > impl.ml
   $ ./extender.exe impl.ml
-  Fatal error: exception (Failure "A raised exception")
+  Fatal error: exception (context-free ppx pass) (Failure "A raised exception")
   [2]
   $ ./extender.exe -embed-errors impl.ml
-  Fatal error: exception (Failure "A raised exception")
+  Fatal error: exception (context-free ppx pass) (Failure "A raised exception")
   [2]
 
  In the case of derivers
 
   $ echo "type a = int" > impl.ml
   $ echo "type b = int [@@deriving deriver_raised_exception]" >> impl.ml
   $ ./deriver.exe -embed-errors impl.ml
-  Fatal error: exception (Failure "A raised exception")
+  Fatal error: exception (context-free ppx pass) (Failure "A raised exception")
```

Comment:
Here it would also be nice to have the suffix "The use of ... is discouraged...", wouldn't it?

Reply:
> so is it really discouraged?

That's a good question. So on one hand, you're right that from merlin's point of view throwing any exception inside a context-free rule is not a big deal if we catch it and embed it by locating it around the node that's being rewritten.  On the other hand, it's also good to encourage PPX authors to think about the location carefully and to point to the exact point of the error. For example, if the error is due to an unexpected payload, it's nice to point to the exact part of the payload that's unexpected instead of the whole node.

Reply:
I think it should be discouraged indeed. If there is a single error, whether it's us or them doing the wrapping won't change anything that's true. Raising in such a way prevents from reporting several errors at once though.

From a user perspective that means: fix one error in the payload, build, repeat, until there's no more errors. I think we can all agree this isn't the best workflow.

Having other exceptions than `Location.Error` escape the ppx expander should definitely be discouraged. Raising `Location.Error` is a slightly different matter as in some cases, doing the warpping yourself can be a bit painful so if you know you can only report a single error, raising can prove convenient, we don't necessarily want to impose this burden on ppx authors.

Reply:
That being said, I'm not sure this should be part of the error message as it's likely to be read by the ppx users rather than authors. I think pointing to the responsible ppx should be enough. Properly documenting the prefered way to report errors and how and where to report such crashes (i.e. on the ppx issue tracker rather than on ppxlib, we could do that in the github issue pattern btw) would be best.

Also is there a reason why this kind of error are reported as raw sexps rather than nicely worded like the other errors in this cram test?